[T09-3] Picconso

benedictcss opened the pull request:
@chivent @j-lum @ihwk1996 @lancelotwillow @benedictcss

eugenepeh commented on README.adoc:
```diff
@@ -1,40 +1,43 @@
+
 ifdef::env-github,env-browser[:relfileprefix: docs/]
 
-https://travis-ci.org/se-edu/addressbook-level4[image:https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master[Build Status]]
```
instead of removing, you can update them with your own repo link

eugenepeh commented on docs/AboutUs.adoc:
```diff
@@ -11,17 +11,17 @@ We are a team based in the http://www.comp.nus.edu.sg[School of Computing, Natio
+image::jeffry.jpg[width="150", align="left"]
+{empty}[https://github.com/j-lum[github]] [<<jeffry#, portfolio>>]
 
 Role: Project Advisor
```
remember to update the roles too

eugenepeh commented on docs/DeveloperGuide.adoc:
```diff
@@ -820,13 +820,14 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
-* has a need to manage a significant number of contacts
-* prefer desktop apps over other types
-* can type fast
+* needs a qiocl amd easy way to edit images
```
typo?

eugenepeh reviewed (commented):
Just a brief look through. Great work so far.

Some things to note though,
- usually user guide should match the current product, so if its not yet implemented, do mark it with a `(future feature)` or something. 
- You can remove the top navigation bar too as it doesn't fit your product nicely.


Keep up the good work towards v1.1

eugenepeh commented on src/main/java/seedu/address/commons/events/ui/ChangeImageEvent.java:
```diff
@@ -0,0 +1,34 @@
+     * Constructor for ChangeImageEvent
+     *
+     * @param image  Image to replace
+     * @param target The name of the ImageView to target.
```
Can probably name the variables to be more intuitive so that you don't have to further explain?

eugenepeh commented on src/main/java/seedu/address/commons/events/ui/ChangeImageEvent.java:
```diff
@@ -0,0 +1,34 @@
+    public final String target;
+
+    /**
+     * Constructor for ChangeImageEvent
```
this can be omit as its self explanatory.

eugenepeh commented on src/main/java/seedu/address/model/canvas/Canvas.java:
```diff
@@ -0,0 +1,20 @@
+
+    private ArrayList<Layer> layers = new ArrayList<>();
+
+    public Canvas(Image base){
```
missing space before left curly `{`

eugenepeh commented on src/main/java/seedu/address/model/google/GoogleClientInstance.java:
```diff
@@ -0,0 +1,24 @@
+//@@author chivent
+
+/**
+ * Google Client Instance
```
need to provide more detail e.g. what is it for.

eugenepeh reviewed (commented):
Good work, code quality wise seems to be good. Some things, however, need to be updated soon. E.g. restore your Travis CI.

Other things:
* Landing page
   * url needs to be updated
![image](https://user-images.githubusercontent.com/19277206/46603012-66dfbc00-cb24-11e8-96f2-67d2d7ca7663.png)

* Gh-pages
   * Update/remove to match your product:
![image](https://user-images.githubusercontent.com/19277206/46603037-7959f580-cb24-11e8-83c5-6a0cafdab9c5.png)

* README
   * Mock-up looks good
* User-guide 
   * looks good
   * features not implemented yet should be clearly marked as Coming in v2.0
* Developer guide 
   * You can remove Appendix A and G if you want
   * _{More to be added}_  can be removed if no longer applicable
   * `Use case: Editing an image` wheres the app's interaction?
   * Your NFR is a bit closely tied to the features. Perhaps can add a few more generic ones. 
   * Still a lot of `AddressBook` references, which should be refactor or removed.

eugenepeh commented:
@chivent @j-lum @ihwk1996 @lancelotwillow @benedictcss



## Feedback for v1.2

Please refer to my v1.1 comments for some of the things that you might have missed out.

### Photos
* One of your team members photo was not rendered properly.
![image](https://user-images.githubusercontent.com/19277206/47276047-0dc55d00-d5e7-11e8-9699-c80cf16e2857.png)

### Developer Guide
* Facilitated by UserPrefs but the direct interaction is model?
![image](https://user-images.githubusercontent.com/19277206/47276269-706b2880-d5e8-11e8-9be6-ff54b972e005.png)

* Does not seems to match the method signature, is this intended?
![image](https://user-images.githubusercontent.com/19277206/47276284-87aa1600-d5e8-11e8-9b13-5aaf81506aff.png)

* Pros and cons not very clearly highlighted for the second part, also do highlight your selected choice and rationale.
![image](https://user-images.githubusercontent.com/19277206/47276152-c8edf600-d5e7-11e8-9818-93dd67cf1cf6.png)


* Typo? 2 of the methods using exact same name. Also, the name for getting the path of execute file is not very intuitive 
![image](https://user-images.githubusercontent.com/19277206/47276081-45cca000-d5e7-11e8-9071-baf78c5ec245.png)

### Overall
* Introduction in each of your feature can include value that it brings to your product etc.
* Work on making the product user-testable soon

lancelotzty committed:
Merge pull request #177 from lancelotwillow/master

modify UG

codacy-badger committed:
Add Codacy badge

codacy-badger committed:
Add Codacy badge

lancelotzty committed:
Merge pull request #179 from codacy-badger/codacy-badge

Add a Codacy badge to README.adoc

lancelotzty committed:
Merge pull request #179 from codacy-badger/codacy-badge

Add a Codacy badge to README.adoc

lancelotzty committed:
Update README.adoc

lancelotzty committed:
Update README.adoc

Commit:
ignore the package for repo sense

lancelotzty committed:
Update config.json

lancelotzty committed:
Update config.json

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

# Conflicts:
#	src/main/java/seedu/address/logic/commands/ConvertCommand.java
#	src/main/java/seedu/address/logic/commands/ExampleCommand.java

jeffrylum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

# Conflicts:
#	src/main/java/seedu/address/logic/commands/ConvertCommand.java
#	src/main/java/seedu/address/logic/commands/ExampleCommand.java

benedictcss committed:
Merge pull request #169 from j-lum/master

!!!Breaking changes multiple features!!!

benedictcss committed:
Merge pull request #169 from j-lum/master

!!!Breaking changes multiple features!!!

Commit:
remove useless package

Commit:
merge the team master for layer

j-lum committed:
Added comments for Layers and Canvas commands

j-lum committed:
Added more comments for Canvas and Layer and ModelManager methods.

benedictcss committed:
update SelectCommand to OpenCommand

benedictcss committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

benedictcss committed:
Merge pull request #183 from j-lum/master

Added comments for all of my files.

benedictcss committed:
fix checkstyle error

benedictcss committed:
fix checkstyle error

j-lum committed:
Removed the canvas save commmand.
Small bug fixes.

Commit:
add test for imageMagickUtil

Commit:
merge changes from team

benedictcss committed:
Merge pull request #184 from benedictcss/refactorSelectToOpen

Refactor selectCommand to openCommand

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

Fixed minor bugs.

Commit:
first test for travis

Commit:
second approach for passing travis

Commit:
merge the team master

j-lum committed:
Removed the canvas save commmand.
Small bug fixes.

Commit:
third approach for passing travis

j-lum committed:
Remove the extra test for raw

Commit:
forth approach for passing travis

Commit:
fifth approach for passing travis

j-lum committed:
Fixed canvas resizing.

Commit:
modify save command

j-lum committed:
Fixed the author tag for most of my commits.
Removed the raw and canvas save commands.

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into github

j-lum committed:
Minor performance fixes recommended by codacy.

j-lum committed:
Merge pull request #186 from j-lum/master

Improved documentation and misc command removals/changes

j-lum committed:
Added a bunch of tests and refactored some error handling in commands.

Also fixed a large number of warnings.

eugenepeh commented:
Some quick feedback for DG at mid-v1.4:
* No consistency with other diagrams in 3.5
![image](https://user-images.githubusercontent.com/19277206/48274580-9749ab80-e47e-11e8-8bb8-431c45422b74.png)
* Problem areas spotted in diagrams:
![image](https://user-images.githubusercontent.com/19277206/48275027-906f6880-e47f-11e8-8f43-e62d0597b0b1.png)

j-lum committed:
Added a bunch of tests and refactored some error handling in commands.

Also fixed a large number of warnings.

j-lum committed:
Updated the DG and my PPP.

j-lum committed:
Merge pull request #188 from j-lum/master

There's quite a bit of warnings for missing documentation, we might want to jump on that.

Commit:
combine raw command

Commit:
modify convert command

Commit:
fix travis

Commit:
fix travis2

Commit:
fix travis3

Commit:
fix travis4

lancelotzty committed:
Merge pull request #181 from lancelotwillow/master

modify the convert, save, imageMagick, and tests

j-lum committed:
Misc codacy fixes.

Commit:
correct methods naming for tests

ihwk1996 committed:
add test for clear cache
delete unused methods in model

lancelotzty committed:
Merge pull request #189 from j-lum/master

Misc codacy fixes.

Commit:
fix conflict

ihwk1996 committed:
Merge commit '69d089685d2f8423474b1b70f4d6cefaf993faf8' into add-test-clearcache-minor-fixes

# Conflicts:
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/test/java/seedu/address/logic/commands/CdCommandTest.java

lancelotzty committed:
Merge pull request #191 from lancelotwillow/master

change test methods naming

ihwk1996 committed:
checkstyle fixes

ihwk1996 committed:
checkstyle fix 2

ihwk1996 committed:
Merge pull request #192 from ihwk1996/add-test-clearcache-minor-fixes

Add test clearcache minor fixes

j-lum committed:
More codacy fixes.

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into github

j-lum committed:
Merge pull request #193 from j-lum/master

More codacy fixes.

Commit:
DeveloperGuide.adoc: Updating diagrams and content for login, Google Ls and Dl

- Also slightly editing code to remove warnings

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into documentation-update

Commit:
DeveloperGuide.adoc: Updating for Google Download Feature

ihwk1996 committed:
DG updates

ihwk1996 committed:
Merge pull request #194 from ihwk1996/undoredo-DG-changes

DG updates

Commit:
DeveloperGuide.adoc: Removing redundant areas

Commit:
Merge branch 'master' into documentation-update

Commit:
DeveloperGuide.adoc: Adding updates to general portions

- Also slight changes to tests for check style

ihwk1996 committed:
spelling fix

Commit:
modify some test in imageMagick

Commit:
fix checkstyle

benedictcss committed:
update command box to handle multiple tabs

Commit:
fix some codacy

benedictcss committed:
Update UGDG for open command

Commit:
fix checkstyle

benedictcss committed:
update reposense

lancelotzty committed:
Merge pull request #196 from lancelotwillow/master

modify some test in imageMagick

benedictcss committed:
Merge branch 'master' into update-ugdg

benedictcss committed:
merge branch

benedictcss committed:
Merge branch 'update-ugdg' of https://github.com/benedictcss/main into update-ugdg

benedictcss committed:
fix test fail on travis

benedictcss committed:
fix codacy

benedictcss committed:
update commandbox test to fix assertion failure

benedictcss committed:
check travis errors

benedictcss committed:
add line to check travis test

lancelotzty commented:
> Some quick feedback for DG at mid-v1.4:
> 
>     * No consistency with other diagrams in 3.5
>       ![image](https://user-images.githubusercontent.com/19277206/48274580-9749ab80-e47e-11e8-8bb8-431c45422b74.png)
> 
>     * Problem areas spotted in diagrams:
>       ![image](https://user-images.githubusercontent.com/19277206/48275027-906f6880-e47f-11e8-8f43-e62d0597b0b1.png)

what you mean by no consistency? and may i also ask, should i show the part i apply the transformation in the diagram? as that is independent to the model

Commit:
DeveloperGuide.adoc: Formatting and Grammar checks

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into documentation-update

# Conflicts:
#	src/main/java/seedu/address/logic/commands/google/GoogleDlCommand.java
#	src/main/java/seedu/address/logic/commands/google/GoogleLsCommand.java

Commit:
modify UG

Commit:
Merge pull request #195 from chivent/documentation-update

Updating Developer Docs

Commit:
Merge branch 'master' into update-ugdg

Commit:
modify UG2

Commit:
fix conflict

Commit:
fix checkstyle

lancelotzty committed:
Merge pull request #199 from lancelotwillow/master

modify UG

Commit:
DeveloperGuide.adoc: Updating the Appendix

- Also updated version number for release

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into documentation-update

Commit:
Editing files to remove warnings

Commit:
Merge pull request #200 from chivent/documentation-update

Documentation update: Updating the Appendix

Commit:
Merge branch 'master' into update-ugdg

Commit:
Adding tests for coveralls

Commit:
Adding Google tests for coveralls

Commit:
Correcting test for Google

benedictcss committed:
merge branch

benedictcss committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into update-ugdg

# Conflicts:
#	docs/UserGuide.adoc
#	src/test/java/seedu/address/ui/CommandBoxTest.java

Commit:
Adding tests for Google and HistoryListPanel

benedictcss committed:
Merge branch 'update-ugdg' of https://github.com/benedictcss/main into update-ugdg

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc

benedictcss committed:
Merge pull request #198 from benedictcss/update-ugdg

Update ugdg

Commit:
adding test for layer panel

Commit:
Merge pull request #201 from chivent/general-updates

General updates for coveralls

benedictcss committed:
remove gif images from imagelist

benedictcss committed:
update benedictcss.adoc

benedictcss committed:
update DG

benedictcss committed:
update userprefs to improve codacy

benedictcss committed:
update remaining convert to apply

benedictcss committed:
Merge pull request #202 from benedictcss/update-ugdg

Remove .gif images from imageList

Commit:
Reformatting error/success messages for applying transformations and fixing display bugs

- Correctly update layer list on swap
- Correctly update layer list name on add after delete
- Correctly update history list on undo -> add -> redo
- Correct auto-resize command name

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into github

Commit:
Correcting common diagrams and messages

Commit:
Updating general portions of userGuide, readMe, aboutUs and contactUs

Commit:
Merge branch 'master' into general-updates

Commit:
added EOF AboutUs

Commit:
Merge pull request #203 from chivent/general-updates

Reformatting error/success messages for applying transformations and …

Commit:
editing user guide

j-lum committed:
Updated UG further.

Fixed ImageMagickUtil to set the bgcolor correctly, fixed SaveCommand.

j-lum committed:
Forgot the new images.

j-lum committed:
Merged before push, sorry to whoever wrote the quick start.

j-lum committed:
Merge pull request #204 from j-lum/master

Fixed UG along with SaveCommand

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into general-updates

j-lum committed:
More grammar changes in the UG.

j-lum committed:
Merge pull request #205 from j-lum/master

Even more grammar changes in the UG.

Commit:
fix test

Commit:
fix test2

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

lancelotzty committed:
Merge pull request #206 from lancelotwillow/master

fix test

j-lum committed:
Fixed format of png files for export.
Split DG and UG into multiple parts.

j-lum committed:
Minor grammar changes

j-lum committed:
Merge pull request #207 from j-lum/master

Fixed format of png files for export.

Commit:
add test

Commit:
Merge branch 'master' of team

Commit:
modify DG

Commit:
merge my master

Commit:
fix checkstyle

Commit:
fix codacy

Commit:
fix checkstyle

lancelotzty committed:
Merge pull request #208 from lancelotwillow/master

add test for coverall

lancelotzty committed:
Update lancelotwillow.adoc

lancelotzty committed:
Update lancelotwillow.adoc

lancelotzty committed:
Update lancelotwillow.adoc

Commit:
update diagram

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

Commit:
fix checkstyle

Commit:
reformatting ug

lancelotzty committed:
Merge pull request #209 from lancelotwillow/master

update diagram for DG

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into general-updates

# Conflicts:
#	docs/UserGuide.adoc

Commit:
Merge branch 'master' into general-updates

Commit:
Merge pull request #210 from chivent/general-updates

General updates to Developer Guide

ihwk1996 committed:
remove top Addressbook bar

ihwk1996 committed:
Change undoredo diagram

j-lum committed:
Minor grammar changes

j-lum committed:
Removed unused class Test.java.

Refactored LayerAddCommand to remove an unreachable exception.

j-lum committed:
Style fixes.

j-lum committed:
Merge pull request #212 from j-lum/master

Improved test coverage by deleting stuff

j-lum committed:
Fixed tianyang's face

benedictcss committed:
Update sequence diagrams and ugdg

j-lum committed:
Rewrote some of Transformations.

ihwk1996 committed:
Merge pull request #211 from ihwk1996/DG-amendments

Change undoredo diagram

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into github

j-lum committed:
Grammar fixes.

j-lum committed:
Merge pull request #214 from j-lum/master

Misc grammar fixes and rewrites for DG

j-lum committed:
Merge pull request #213 from benedictcss/update-ugdg

Update sequence diagrams and ugdg

j-lum committed:
Fixed comments in code.

j-lum committed:
Merge pull request #215 from j-lum/master

Fixed comments in code.

Commit:
change wording

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

j-lum committed:
Gave `apply raw` it's own section, clarified the `apply resize` command.

ihwk1996 committed:
minor english changes

ihwk1996 committed:
Merge pull request #217 from ihwk1996/ivan-PPP-changes

minor english changes

benedictcss committed:
update benedictcss.adoc

benedictcss committed:
update diagram

j-lum committed:
Fixed the conjugation of canvas size.

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into github

j-lum committed:
Merge pull request #219 from j-lum/master

Extremely important changes to canvas conjugation.

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into github

ihwk1996 committed:
update ihwk1996.adoc

Commit:
Updated own PPP doc.

benedictcss committed:
Merge pull request #218 from benedictcss/update-ugdg

update benedictcss.adoc

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into general-updates

Commit:
update Developer Guide Diagram

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

ihwk1996 committed:
update ivan page

ihwk1996 committed:
increase photo size

lancelotzty committed:
Merge pull request #216 from lancelotwillow/master

change wording

ihwk1996 committed:
Merge pull request #221 from ihwk1996/ivan-PPP-more-changes

Ivan ppp more changes

Commit:
Merge pull request #220 from chivent/general-updates

Updated own adoc.

ihwk1996 committed:
update DG with figure labels

benedictcss committed:
update ppp benedictcss

ihwk1996 committed:
Merge pull request #223 from ihwk1996/ivan-general-updates

update DG with figure labels

ihwk1996 committed:
typo fix

Commit:
add image

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main

lancelotzty committed:
Merge pull request #224 from lancelotwillow/master

add image for DG

lancelotzty committed:
Update lancelotwillow.adoc

Commit:
fix image

benedictcss committed:
update adocs to include manual testing

Commit:
fix image2

benedictcss committed:
further updates on manual testing in DG

lancelotzty committed:
Merge pull request #226 from lancelotwillow/master

fix image

benedictcss committed:
Merge pull request #225 from benedictcss/update-ugdg

Update DG to include Manual testing

lancelotzty committed:
Update lancelotwillow.adoc

j-lum committed:
Only known image of me in a suit.

j-lum committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into github

j-lum committed:
Only known image of me in a suit.

benedictcss committed:
update tags in DG

benedictcss committed:
fix tags in DG

benedictcss committed:
Merge pull request #227 from benedictcss/update-ugdg

Update ugdg

benedictcss committed:
update benedictcss.adoc

Commit:
add manual test

j-lum committed:
Updated the DG to include a testing guide.

lancelotzty committed:
Merge pull request #228 from lancelotwillow/master

add manual test

j-lum committed:
Merge pull request #229 from j-lum/master

Updated the DG to include a testing guide.

benedictcss committed:
update own adoc

benedictcss committed:
update my adoc

j-lum committed:
Delete j-lum.pdf

benedictcss committed:
update ascii.adoc

ihwk1996 committed:
update manual testing of undo redo

ihwk1996 committed:
Merge pull request #230 from ihwk1996/undoredo-manual-tests

update manual testing of undo redo

Commit:
developer and user guide update

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-3/main into general-updates

# Conflicts:
#	docs/DeveloperGuide.adoc

Commit:
Merge pull request #231 from chivent/general-updates

General updates

j-lum committed:
Pants wetting exception fixing experience.

j-lum committed:
More fixes.

j-lum committed:
Please figure out how to write tests.

j-lum committed:
Merge pull request #232 from j-lum/master

Pants wetting exception fixing experience.

j-lum committed:
Please figure out how to write tests.

j-lum committed:
Merge pull request #233 from j-lum/master

Please figure out how to write tests.

lancelotzty committed:
Update SaveCommandTest.java

j-lum committed:
Clued in clueless users.
Change the percentage in resize to match code.
Reminded users that layer and canvas operations are in fact **NOT** transformations.
Change `layer remove` to `layer delete`.
Added notice that positions will not take anything longer than a int.
Copied tutorial images out so that help will display them.

j-lum committed:
Added blurb about rotation center and added missing image files.

j-lum committed:
Added alternate link for tutorial.

j-lum committed:
Added a blurb about file formats supported.

Commit:
added warning for redirecting

Commit:
removal of google auto-login since it does not seem to add value to users

Commit:
check style fixes

Commit:
Merge pull request #235 from chivent/master

removal of auto-login
